packages/atomic/src/utils/initialization-utils.tsx

```diff
@@ -290,4 +290,26 @@ export function DeferUntilRender() {
   };
 }
 
+export function RemainHiddenUntilDescendantsRendered() {
+  return (component: ComponentInterface, methodName: string) => {
+    const {componentDidRender} = component;
+
+    component[methodName] = function () {
+      const element = getElement(this);
+      if (!element.classList.contains('atomic-hidden')) {
```

Comment:
I'm a little confused, couldnt this be achieved with css only?

Reply:
No, because the element loses `atomic-hidden`. We're basically extending the life of `atomic-hidden`, but using `visibility: hidden` instead so we can estimate the size of children.